torch/_cuda_sanitizer/_sanitizer.py

```diff
@@ -0,0 +1,490 @@
+import enum
+import functools
+import io
+import logging
+import sys
+import textwrap
+import traceback
+from dataclasses import dataclass, field
+from typing import Any, Dict, Iterator, List, Optional, Set, Tuple, TypeVar
+
+import torch
+import torch.utils._cuda_trace as cuda_trace
+from torch.utils._cuda_trace import _CUDA_HOOK_TRACE_OFFSET
+from torch.utils._python_dispatch import (
+    TorchDispatchMode,
+    _PYTHON_DISPATCH_TRACE_OFFSET,
+)
+from torch.utils._pytree import tree_map
+
```

Comment:
Would be nice to have a top level comment describing the overall strategy for this file